packages/core/src/system.ts

```diff
@@ -190,17 +201,14 @@ export function createSystem<T extends ActorSystemInfo>(
       reverseKeyedActors.set(actorRef, systemId);
     },
     inspect: (observer) => {
-      observers.add(observer);
-    },
-    _sendInspectionEvent: (event) => {
-      const resolvedInspectionEvent: InspectionEvent = {
-        ...event,
-        rootId: rootActor.sessionId
-      };
-      observers.forEach((observer) => observer.next?.(resolvedInspectionEvent));
+      inspectionObservers.add(observer);
+
+      // Only send inspection events if someone is listening
+      system._sendInspectionEvent = sendInspectionEvent as any;
```

Comment:
It would make sense to make `system._sendInspectionEvent` non-optional and return early there based on `!inspectionObservers.size`